### PR TITLE
[otbn] fix issues in simulator

### PR DESCRIPTION
**Pull request by felixmiller**
Some issues I came across when running RSA on the new simulator:

- shift amount in load method in dmem model (see #3505)
- operand oder with bitwise OR (see #3506)
- register field names for `CSR` instructions (see #3507)
- order of flag names in lists (#3508)
- list iteration in dmem model for commit method (see #3509)

### CODE REVIEW

File: hw/ip/otbn/dv/otbnsim/sim/insn.py [hunk 1]
```diff
@@ -810,7 +810,7 @@ def __init__(self, op_vals: Dict[str, int]):
     def execute(self, model: OTBNModel) -> None:
         idx = self.wsr
         old_val = model.state.wcsr_read(idx)
-        new_val = old_val | model.state.wreg[self.wrs]
+        new_val = model.state.wreg[self.wrs] | old_val
```
Comment (rswarbrick) [inline hw/ip/otbn/dv/otbnsim/sim/insn.py:813]: I don't think we actually need `new_val` to be a register. Can we just call `model.state.wreg[self.wrs].unsigned()` instead? It seems a lot less magic.

**Review by rswarbrick — COMMENTED** [review]
Hi Felix! Sorry for all the clashing PRs. I think that most of these changes are now addressed elsewhere, and the following remain:
- [otbn] fix register names in simulator
- [otbn] change order of operands in simulator

Does that sound right to you?

**Comment by imphil** [comment]
How much of that is still relevant if #3578 goes in @rswarbrick? It looks like most (all?) changes are addressed by your PR as well.

**Comment by rswarbrick** [comment]
Yes, I *think* the remaining parts of this are now done. @felixmiller: could you do a quick check to make sure I haven't missed anything?

**Comment by felixmiller** [comment]
@rswarbrick No worries, good that it is addressed elsewhere

@imphil and @rswarbrick: yes they all seem to be addressed. I cannot reproduce my results though with current state of master, but it seems to be a different issue (it seems that dmem is populated with incorrect data).

Closing this PR now and will close the individual issues once I am able to reproduce.

**Pull request closed without merging**